src/economy/economy.cpp

```diff
@@ -124,116 +157,125 @@ void initialize_artisan_distribution(sys::state& state) {
 	}
 }
 
-inline constexpr float lx_extra_factor = 6.0f;
-inline constexpr float en_extra_factor = 2.5f;
+
+void initialize_needs_weights(sys::state& state, dcon::nation_id n) {
+	{
+		state.world.for_each_commodity([&](dcon::commodity_id c) {
+			auto kf = state.world.commodity_get_key_factory(c);
+			if(state.world.commodity_get_is_life_need(c) && (state.world.commodity_get_is_available_from_start(c) || (kf && state.world.nation_get_active_building(n, kf)))) {
+				auto& w = state.world.nation_get_life_needs_weights(n, c);
+				w = 1.f;
+			}
+		});
+	}
+
+	{
+		state.world.for_each_commodity([&](dcon::commodity_id c) {
+			auto kf = state.world.commodity_get_key_factory(c);
+			if(state.world.commodity_get_is_everyday_need(c) && (state.world.commodity_get_is_available_from_start(c) || (kf && state.world.nation_get_active_building(n, kf)))) {
+				auto& w = state.world.nation_get_everyday_needs_weights(n, c);
+				w = 1.f;
+			}
+		});
+	}
+
+	{
+		state.world.for_each_commodity([&](dcon::commodity_id c) {
+			auto kf = state.world.commodity_get_key_factory(c);
+			if(state.world.commodity_get_is_luxury_need(c) && (state.world.commodity_get_is_available_from_start(c) || (kf && state.world.nation_get_active_building(n, kf)))) {
+				auto& w = state.world.nation_get_luxury_needs_weights(n, c);
+				w = 1.f;
+			}
+		});
+	}
+}
+
+float need_weight(sys::state& state, dcon::nation_id n, dcon::commodity_id c) {
+	return 1 / sqrt(state.world.commodity_get_current_price(c));
```

Comment:
this may potentially be an issue down the line. Unless you know otherwise, my understanding is that the result of the sqrt function is not defined down to the last bit of precision between different compilers. To keep deterministic lockstep working, you probably need to use `math::sqrt` which, with the implementation we provide, should always give the same results, no matter where it is compiled.